Add RUSTSTDLIB variable to configure

Firefox expects rust to work from standard install, which is not the
case for OE cross compiled rust. We need to suggest the path for
rustlibs using -L option

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Khem Raj <raj.khem@gmail.com>

Index: firefox-60.1.0/build/moz.configure/rust.configure
===================================================================
--- firefox-60.1.0.orig/build/moz.configure/rust.configure
+++ firefox-60.1.0/build/moz.configure/rust.configure
@@ -202,14 +202,22 @@ option(env='RUST_TARGET',
 def rust_target(value):
     return value[0]
 
-@depends(target, rustc, rust_target, when=rust_compiler)
+option(env='RUSTSTDLIB',
+       nargs=1,
+       help='Rust std lib')
+@depends('RUSTSTDLIB')
+@checking('rust std lib', lambda target: target)
+def ruststdlib(value):
+    return value[0]
+
+@depends(target, rustc, rust_target, ruststdlib, when=rust_compiler)
 @imports('os')
 @imports('subprocess')
 @imports(_from='mozbuild.configure.util', _import='LineIO')
 @imports(_from='mozbuild.shellutil', _import='quote')
 @imports(_from='tempfile', _import='mkstemp')
 @imports(_from='textwrap', _import='dedent')
-def available_rust_target(target, rustc, rust_target):
+def available_rust_target(target, rustc, rust_target, ruststdlib):
     # Check to see whether our rustc has a reasonably functional stdlib
     # for our chosen target.
     target_arg = '--target=' + rust_target
@@ -228,6 +236,7 @@ def available_rust_target(target, rustc,
         cmd = [
             rustc,
             '--crate-type', 'staticlib',
+            '-L', ruststdlib,
             target_arg,
             '-o', out_path,
             in_path,
